pkg/open/open.go

```diff
@@ -198,7 +198,12 @@ func (o *opener) pod(ctx context.Context, service *corev1.Service) (*corev1.Pod,
 	if len(pods.Items) < 1 {
 		return nil, fmt.Errorf("no pod found for service %v", service.Name)
 	}
-	return &pods.Items[0], nil
+	for _, pod := range pods.Items {
+		if isPodReady(pod) {
```

Comment:
This could be simplified with `IsStatusConditionTrue(pod.Status.Conditions, corev1.ConditionTrue)` from `k8s.io/apimachinery/pkg/api/meta/conditions.go`

Reply:
Unfortunately not. I was surprised too, but `corev1.PodCondition` is distinct from `metav1.Condition`